eox_core/api/support/v1/tests/integration/test_views.py

```diff
@@ -0,0 +1,298 @@
+        response_data = response.json()
+        get_response = self.get_user(self.tenant_x, {"email": data["email"]})
+        get_response_data = get_response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response_data, f"The user {data['username']} <{data['email']}>  has been removed")
+        self.assertEqual(get_response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            get_response_data,
+            {"detail": f"No user found by {{'email': '{data['email']}'}} on site {self.tenant_x['domain']}."},
+        )
+
+    @ddt.data(
+        ("username", "user-not-found"),
+        ("email", "user-not-found@mail.com"),
+    )
+    @ddt.unpack
+    def test_delete_user_in_tenant_not_found(self, query_param: str, value: str) -> None:
+        """
+        Test delete an edxapp user in a tenant that does not exist.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+
+        Expected result:
+        - The status code is 404.
+        - The response indicates the user was not found in the tenant.
+        """
+        response = self.delete_user(self.tenant_x, {query_param: value})
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            response_data["detail"],
+            f"No user found by {{'{query_param}': '{value}'}} on site {self.tenant_x['domain']}.",
+        )
+
+    def test_delete_user_missing_required_fields(self) -> None:
+        """
+        Test delete an edxapp user in a tenant without providing the username or email.
+
+        Expected result:
+        - The status code is 400.
+        - The response indicates the username or email is required.
+        """
+        response = self.delete_user(self.tenant_x)
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_data, ["Email or username needed"])
+
+    @ddt.data("username", "email")
+    def test_update_username_in_tenant_success(self, query_param: str) -> None:
+        """
+        Test update an edxapp user's username in a tenant.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `check_edxapp_account_conflicts`
+        - `replace_username_cs_user`
+        - `get_user_read_only_serializer`
+
+        Expected result:
+        - The status code is 200.
+        - The response indicates the username was updated successfully.
+        - The user is found in the tenant with the new username.
+        """
+        data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, data)
+        new_username = f"new-username-{query_param}"
+
+        response = self.update_username(self.tenant_x, {query_param: data[query_param]}, {"new_username": new_username})
```

Comment:
Same question here about updating a user from a different tenant. Should that work?